packages/neon/neon_talk/lib/src/widgets/message.dart

```diff
@@ -356,24 +362,94 @@ class TalkCommentMessage extends StatelessWidget {
   /// {@macro TalkMessage.isParent}
   final bool isParent;
 
+  @override
+  State<TalkCommentMessage> createState() => _TalkCommentMessageState();
+}
+
+class _TalkCommentMessageState extends State<TalkCommentMessage> {
+  BuiltMap<String, Result<core.Reference>> references = BuiltMap();
+  StreamSubscription<Result<RegExp?>>? referenceRegexSubscription;
+  StreamSubscription<BuiltMap<String, Result<core.Reference>>>? referencesSubscription;
+
+  @override
+  void initState() {
+    super.initState();
+
+    if (!widget.isParent) {
+      const kMaxReferences = 3;
+
+      final referencesBloc = NeonProvider.of<ReferencesBloc>(context);
+
+      referenceRegexSubscription = referencesBloc.referenceRegex.listen((result) {
+        final referenceRegex = result.data;
+        if (referenceRegex == null) {
+          return;
+        }
+
+        final matches = referenceRegex
+            .allMatches(widget.chatMessage.message)
+            .map((match) => match.group(0)!.trim())
+            .take(kMaxReferences)
+            .toBuiltList();
+
+        setState(() {
+          references = references.rebuild((b) {
+            for (final match in matches) {
+              b[match] ??= Result.loading();
+            }
+
+            b.removeWhere((key, value) => !matches.contains(key));
+          });
+        });
+
+        if (matches.isNotEmpty) {
+          referencesBloc.loadReferences(matches);
+        }
+      });
+
+      referencesSubscription = referencesBloc.references.listen((result) {
+        if (!context.mounted) {
+          return;
+        }
+
+        setState(() {
+          references = references.rebuild((b) {
+            for (final url in references.keys) {
+              b[url] = result[url] ?? Result.loading();
+            }
+
+            b.removeWhere((key, value) => !result.keys.contains(key));
+          });
+        });
+      });
```

Comment:
I think that this logic should be in the bloc

Reply:
I wanted to do that at first, but it doesn't fit in the ReferencesBloc. I could make a bloc per message, but I'm not sure if that is a good idea.